apps/dashboard/src/components/side-navigation/usage-card.tsx

```diff
@@ -0,0 +1,120 @@
+import { useTelemetry } from '@/hooks/use-telemetry';
+import { ROUTES } from '@/utils/routes';
+import { TelemetryEvent } from '@/utils/telemetry';
+import { GetSubscriptionDto } from '@novu/shared';
+import { format } from 'date-fns';
+import { RiCalendarEventLine, RiErrorWarningLine } from 'react-icons/ri';
+import { Link } from 'react-router-dom';
+import { Button } from '../primitives/button';
+import { Progress } from '../primitives/progress';
+
+type UsageStatus = {
+  progressVariant: 'error' | 'warning' | 'default';
+  isComplete: boolean;
+};
+
+export type UsageCardProps = {
+  subscription?: GetSubscriptionDto;
```

Comment:
Consider making this required, the component won't load before the subscription is loaded so all the conditionals and optional chaining inside the component will go away.